pkg/sql/catalog/lease/lease_test.go

```diff
@@ -925,6 +920,10 @@ func TestDescriptorRefreshOnRetry(t *testing.T) {
 	defer srv.Stopper().Stop(context.Background())
 	s := srv.ApplicationLayer()
 
+	// Disable stats collection, which may acquire this descriptor.
+	if _, err := sqlDB.Exec("SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false"); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
we think the issue is the background stats collection job select from the table and thus need to acquire a lease on the table. This would make the check below (acquire = release+1) fail. This is a plausible explanation.

What we don't understand is why
1). this only happens in CI, but not locally (even with --stress)
2). this test passes before this change (without disabling stats collection).

We should try to understand and answer those two questions